sql/src/planner.rs

```diff
@@ -353,112 +478,67 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
 
         // Tsid column is a reserved column.
         ensure!(
-            !name_column_map.contains_key(TSID_COLUMN),
+            !columns_by_name.contains_key(TSID_COLUMN),
             ColumnNameReserved {
                 name: TSID_COLUMN.to_string(),
             }
         );
 
-        // Find timestamp key and primary key contraint
-
-        let mut timestamp_column_idx = None;
-        let mut timestamp_name = None;
-
-        let mut primary_key_column_idxs = vec![];
-
-        for constraint in stmt.constraints.iter() {
-            if let TableConstraint::Unique {
-                columns,
-                is_primary,
-                ..
-            } = constraint
-            {
-                if *is_primary {
-                    // Build primary key, the builder will check timestamp column is in primary key.
-                    for column in columns {
-                        if let Some(idx) = name_column_index_map.get(&*column.value) {
-                            primary_key_column_idxs.push(*idx);
+        let timestamp_column =
+            Self::find_and_ensure_timestamp_column(&columns_by_name, &stmt.constraints)?;
+        let tsid_column = Ident::with_quote(DEFAULT_QUOTE_CHAR, TSID_COLUMN);
+        let mut columns: Vec<_> = stmt.columns.iter().map(|col| col.name.clone()).collect();
+        let mut enable_tsid_primary_key = false;
+
+        let mut add_tsid_column = || {
+            columns_by_name.insert(TSID_COLUMN, Self::tsid_column_schema()?);
+            column_idxs_by_name.insert(TSID_COLUMN, columns.len());
+            columns.push(tsid_column.clone());
+            enable_tsid_primary_key = true;
```

Comment:
I dont remember why this field is required, maybe can be removed?

Reply:
Let's remove it in the next PR because this needs updating of ceresdbproto.